lib/tasks/dev/migrate_title_data.rake

```diff
@@ -0,0 +1,28 @@
+
+namespace :dev do
+   
+    desc 'migrate Student.title data to Title.name column and assign reference id'
+
+    task migrate_title_data: :environment do
+
+        # populate Title.name data with unique old_title values 
+        Student.uniq.pluck(:old_title).each do |title|
+            t = Title.find_or_create_by_name(title) unless !title.present?
+            t.save!
+        end
+        
+        # assign title_id to each student in batches 
+        Title.all.each do |title|
+            Student.where(:old_title => title.name).find_in_batches do |students|
```

Comment:
@Luke-kb .. in general we prefer not to the style with hash rockets =>
Prefer
```
Student.where(:old_title => title.name).....
```

Reply:
Would the `update_all` refactor still be appropriate in a 1mil+ student db?